pkg/monitoring/metrics/virt-handler/migrationdomainstats/handler.go

```diff
@@ -0,0 +1,95 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright the KubeVirt Authors.
+ */
+
+package migrationdomainstats
+
+import (
+	"sync"
+
+	"k8s.io/client-go/tools/cache"
+	v1 "kubevirt.io/api/core/v1"
+
+	"kubevirt.io/kubevirt/pkg/controller"
+)
+
+type handler struct {
+	vmiStore cache.Store
+	vmiStats map[string]*queue
+
+	mutex *sync.Mutex
```

Comment:
- in our use case we expect to have just 1 reader 

- `Collect()` can't be a read-lock as it also changes the data struct to delete the key from a finished migration. I thought about read-locking when creating the output result and then a write-lock to remove the key, but this wouldn't work well because the `isActive` value can become stale between the reading and the write-lock.

Reply:
question: same for the `queue` struct?

Reply:
updated for queue too